### PR TITLE
Hands

### PR DESCRIPTION
**Pull request by cwyoder**
Users can add cards to their hand by dragging to the bottom of the table.

gameState should adjust along with changes, so cards in other users hands will exist in the table upon a user joining, but they cannon be seen.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const { setSocketServer }  = require('./socketServer');
 
 const app = express();
 
-app.use(morgan('dev'));
+// app.use(morgan('dev'));
```
Comment (adam-adair) [inline server/index.js]: I think we want to keep the logs on dev

File: server/socketServer.js [hunk 2]
```diff
@@ -23,6 +23,14 @@ const setSocketServer = (server) => {
             if(cardState.isDragging) this.emit('receiveCard', {...cardState.card, otherPlayerDragging: !cardState.otherPlayerDragging});
         })
 
+        socket.on('addCardToHand', function(cardState) {
```
Comment (adam-adair) [inline server/socketServer.js:25]: you could destructure this if you wanna

File: src/Card.js [hunk 3]
```diff
@@ -75,25 +80,72 @@ export default class Card extends Phaser.GameObjects.Container {
...
+        const dy = (lastY - penY) * 50;
+        this.body.setVelocity(dx, dy);
+      } else this.body.setVelocity(0,0)
+      this.x = dragX;
+      this.y = dragY;
+    }
+
+    switch(this.playerNumber) {
+      case 2:
+        this.addToHand = dragX > canvasWidth - 100;
```
Comment (adam-adair) [inline src/Card.js]: maybe should have 100 in constants so we can change dimensions if we change table size

File: src/DeckrTable.js [hunk 4]
```diff
@@ -93,6 +99,8 @@ export class DeckrTable extends Phaser.Game {
         const spinSpeed = Math.sqrt(((avx+bvx)**2) + ((avy+bvy)**2))
         chipA.setAngularVelocity(1 * (spinCoeff) * spinSpeed * -1)
         chipB.setAngularVelocity(1 * (spinCoeff) * spinSpeed)
+        // socket.emit('sendChip', { chip: chipA, room: gameState.room, otherPlayerDragging: false })
```
Comment (adam-adair) [inline src/DeckrTable.js]: Never mind, I see now that this is leftover from testing the chip physics, just delete before merging this PR

File: src/views/table.js [hunk 5]
```diff
@@ -57,6 +57,18 @@ async function renderTable(tableNumber,playerNumber) {
           <button id="newCard">Deal a card (0)</button>
           <button id="collectCards">Collect cards & shuffle</button>
         </p>
+        <p>
```
Comment (adam-adair) [inline src/views/table.js]: Is this used anywhere? I assume we want to show the number of cards in players hands. I didn't find it, but maybe I missed somethig.

**Review by adam-adair — CHANGES_REQUESTED** [review]
Left a comment to this effect, but I couldn't get hands to show between players

I found a few bugs testing this:
When I have a card in hand as P2, and I deal a card as P1, my P2 card in hand rotates.
I'm not 100% sure how to recreate this, but when I take a card out of my hand as a player and then try to move it as another player, the world bounds aren't quite right. I'm able to take it off the screen sometimes.

**Comment by cwyoder** [comment]
I think I fixed all of the errors/bugs, except for one with stacks ... cards in hand do not change their depth order when moving them in your hand, which makes it harder to reorganize them in your hand. Also, the stack button covers up the card number while it's in your hand.

My opinion on this is that cards in hand should not be able to be stacked, so that stack button shouldn't show up in there at all. I'm free all day today to pair on this.

**Review by adam-adair — APPROVED** [review]
Looks good.

**Pull request merged**